server.js

```diff
@@ -765,12 +766,44 @@ cache(function(data, match, sendBadge, request) {
     }
     try {
       var data = JSON.parse(buffer);
-      // Grab the latest stable version, or an unstable
+
       var versions = Object.keys(data.package.versions);
-      var version = latestVersion(versions);
-      var vdata = versionColor(version);
-      badgeData.text[1] = vdata.version;
-      badgeData.colorscheme = vdata.color;
+      var stableVersion = latestVersion(versions);
+      var vdata = versionColor(stableVersion);
+
+      var badgeText = null;
+      var badgeColor = null;
+
+      switch (info) {
+          case 'v':
+              badgeText = vdata.version;
+              badgeColor = vdata.color;
+              break;
+          case 'vpre':
+              if (versions[0] !== undefined) {
```

Comment:
Ah, I see. Yes, checking the length will be clearer.

On Wed, Feb 11, 2015 at 12:28 AM, Wouter De Schuyter <
notifications@github.com> wrote:

> In server.js
> <https://github.com/badges/shields/pull/366#discussion_r24461635>:
>
> > -      var vdata = versionColor(version);
> > -      badgeData.text[1] = vdata.version;
> > -      badgeData.colorscheme = vdata.color;
> > +      var stableVersion = latestVersion(versions);
> > +      var vdata = versionColor(stableVersion);
> > +
> > +      var badgeText = null;
> > +      var badgeColor = null;
> > +
> > +      switch (info) {
> > +          case 'v':
> > +              badgeText = vdata.version;
> > +              badgeColor = vdata.color;
> > +              break;
> > +          case 'vpre':
> > +              if (versions[0] !== undefined) {
>
> Well, I suppose line 771 - 772 can be moved inside of the vpre case if
> that's what you mean?
>
> L783 is just a habbit, checking if it exists, but might be cleaner to
> change that to if (versions) or if (versions.length > 0).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/badges/shields/pull/366/files#r24461635>.
>